Islandora/islandora_paged_content
# ISLANDORA-2004: Predicates to handle alternate viewing options for paged content models

https://jira.duraspace.org/projects/ISLANDORA/issues/ISLANDORA-2004

# What does this Pull Request do?

Defines 2 new predicates for the paged Solution pack, to be able to define:

 

1. A, per Islandora Object, viewing hint stating if the sequence is side by side (a.k.a as `paged`) or continuous single pages. 
2. If the book has or not a cover page, changing this way the order the first recto is displayed (or the other way around if the progression is Right to left). This is needed for pages that start with a 2 page spread and don't include cover and/or first recto. Those books right now are displayed wrong and with an offset always of 1 page.

This is a closer approach to what IIIF defines for their `sc:Sequence` `@type` where predicates to handle this type of scenarios exist. See http://iiif.io/api/presentation/2.1/#viewinghint

The cover-less book is an addition based on a real use case, found in books that belong to a larger compendium, bound together with other books, but that should be handled as individual objects, so no cover and not first recto. But can be used to display any 2-page spread by default.

# What's new?

New predicates in RELS-EXT for content models that use pages and very bad wording in some new Forms that where added to manage->book

# How should this be tested?
1. Please apply this pull request plus
2. Apply also https://github.com/Islandora/islandora_internet_archive_bookreader/pull/108
3. Go to a book Object of your liking, look at it, then press on `manage->book`
or go directly to `islandora/object/YOURPID/manage/book`
4. After the `Page progression` option there are two new ones:
![cover](https://user-images.githubusercontent.com/6946023/28171621-07ac018e-67b7-11e7-92fb-b3887b133e93.jpeg)
![hint](https://user-images.githubusercontent.com/6946023/28171622-07b17af6-67b7-11e7-9416-2e5b4d196434.jpeg)

Test both and see how your book displays.

5. Since the options could possibly match your saved  Internet Archive Book Reader global ones, e.g This `Paged` is equivalent to Internet Archive Book Reader's 2-up (in that case you will see no change at all!),so  try, if possible playing with that, changing global to `1-up`  and setting `Paged`  for your book. Then compare to other books. See how global is respected. Then do it the other way around. `Without cover` has no global equivalent in islandora, so testing it is easier. Setting it will make your book start with a 2-page spread (if `Paged` was set of course!)
5. Please: proofread my form wording, it is terrible! I know..I could need some help there 😬 

# Additional Notes:

- Wording. The changes are really small, but the Form wording that I used for the new options don't seem to be the best.

- If this change gets merged, then I will also update documentation, screenshots and will add the new predicates to the Islandora Ontology.

* Does this change require documentation to be updated? 
Yes. It adds new options (form) when managing pages. Probably will need some help with the wording in the Forms that were added!

* Does this change add any new dependencies? 
Not by itself, but to really make sense a sister pull request will be done on internet archive reader

* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? 
No, the existing defaults are respected, so if not needed won't touch any objects. All changes could be applied selectively to Objects that could need this by a responsible human being. So no under the hood updates to your precious assets.

* Could this change impact execution of existing code?
No. No weird or heavy logic here.

# Interested parties
Book addicts, @Islandora/7-x-1-x-committers, @adam-vessey @jordandukart @rosiel
